R/init.R

```diff
@@ -136,6 +139,10 @@ init <- function(data,
     modules <- do.call(teal::modules, modules)
   }
 
+  landing <- extract_landing(modules)
+  if (length(landing) > 1L) stop("teal only supports apps with one module of `tm_landing_poup` class.")
```

Comment:
if we go with a more general functionality discussed [above](https://github.com/insightsengineering/teal/pull/934/files#r1368538617) then why restriction on the only one element of such class? There was one question in the discussion that could be avoided - [link](https://github.com/insightsengineering/teal/pull/934#issuecomment-1771274945)

Reply:
As for full generalization, I am with @gogonzo, let's not abstract for use cases that we don't yet have if the changes to `teal` that we are planning will make those potential use cases easier to satisfy.

Reply:
leaning towards what chlebowa and gogonzo mentioned as well